system/autoware_error_monitor/src/autoware_error_monitor_core.cpp

```diff
@@ -13,24 +13,59 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#define FMT_HEADER_ONLY
+#include <fmt/format.h>
 
-#include <autoware_error_monitor/autoware_error_monitor_node.h>
+#include <autoware_error_monitor/autoware_error_monitor_core.hpp>
 
-#include <fmt/format.h>
+AutowareErrorMonitor::AutowareErrorMonitor()
+: Node("autoware_error_monitor"),
+  update_rate_(declare_parameter("update_rate", 10))
+{
+  // Parameter
+  // update_rate_ = declare_parameter("update_rate");
+  loadRequiredConditions(KeyName::manual_driving);
+  loadRequiredConditions(KeyName::autonomous_driving);
+  loadRequiredConditions(KeyName::remote_control);
+  loadRequiredConditions(KeyName::safe_stop);
+  loadRequiredConditions(KeyName::emergency_stop);
+
+  // Subscriber
+  sub_diag_array_ = create_subscription<diagnostic_msgs::msg::DiagnosticArray>(
+    "input/diag_array", rclcpp::QoS{1},
+    std::bind(&AutowareErrorMonitor::onDiagArray, this, std::placeholders::_1));
+
+  // Publisher
+  pub_driving_capability_ = create_publisher<autoware_system_msgs::msg::DrivingCapability>(
+    "output/driving_capability", rclcpp::QoS{1});
+
+  // Timer
+  auto timer_callback = std::bind(&AutowareErrorMonitor::onTimer, this);
+  auto period = std::chrono::duration_cast<std::chrono::nanoseconds>(
+    std::chrono::duration<double>(update_rate_));
```

Comment:
You have to take an inverse of rate in order to get period.
```suggestion
    std::chrono::duration<double>(1.0 / update_rate_));
```

Reply:
Thanks, I missed the conversion on this one!